plugins/nexus-repository-conda/src/main/java/org/sonatype/nexus/repository/conda/internal/hosted/CondaPath.java

```diff
@@ -0,0 +1,277 @@
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.conda.internal.hosted;
+
+import com.google.common.collect.ImmutableList;
+import org.sonatype.nexus.common.hash.HashAlgorithm;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sonatype.nexus.repository.conda.internal.util.CondaPathUtils.REPODATA_JSON;
+
+public class CondaPath {
+
+    public enum HashType {
+        SHA1("sha1", HashAlgorithm.SHA1),
+        MD5("md5", HashAlgorithm.MD5);
+
+        /**
+         * {@link HashAlgorithm}s corresponding to {@link HashType}s.
+         */
+        public static final List<HashAlgorithm> ALGORITHMS = ImmutableList.of(SHA1.getHashAlgorithm(), MD5.getHashAlgorithm());
+
+        private final String ext;
+
+        private final HashAlgorithm hashAlgorithm;
+
+        HashType(final String ext, final HashAlgorithm hashAlgorithm) {
+            this.ext = ext;
+            this.hashAlgorithm = hashAlgorithm;
+        }
+
+        public String getExt() {
+            return ext;
+        }
+
+        public HashAlgorithm getHashAlgorithm() {
+            return hashAlgorithm;
+        }
+    }
+
+
+    public static class Coordinates {
+        private final String packageName;
+
+        private final String version;
+
+        private final String buildString;
+
+        private final String extension;
+
+
+        public Coordinates(final String packageName,
+                           final String version,
+                           final String buildString,
+                           final String extension) {
+            this.packageName = checkNotNull(packageName);
+            this.version = checkNotNull(version);
+            this.buildString = checkNotNull(buildString);
+            this.extension = checkNotNull(extension);
+        }
+
+        @Nonnull
+        public String getPackageName() {
+            return packageName;
+        }
+
+        @Nonnull
+        public String getVersion() {
+            return version;
+        }
+
+        @Nonnull
+        public String getBuildString() {
+            return buildString;
+        }
+
+        @Nonnull
+        public String getExtension() {
+            return extension;
+        }
+    }
+
+
+    private final String path;
+
+    private final String fileName;
+
+    private final Optional<HashType> hashType;
+
+    private final Optional<Coordinates> coordinates;
+
+    public static CondaPath build(String path) {
+        Optional<Coordinates> coordinates = parseCoordinates(path);
+        checkNotNull(path);
```

Comment:
`checkNotNull` should be the first line of the method

Reply:
it should not be here at all :) the optional concept is here for it. Good feedback - thank you